apis/kueue/v1beta2/clusterqueue_types.go

```diff
@@ -202,30 +211,34 @@ type ResourceGroup struct {
 	// The list cannot be empty and it can contain up to 64 flavors, with a max of
 	// 256 total flavors across all resource groups in the ClusterQueue.
 	// +listType=map
+	// +required
 	// +listMapKey=name
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=64
-	Flavors []FlavorQuotas `json:"flavors"`
+	Flavors []FlavorQuotas `json:"flavors,omitempty"`
 }
 
 type FlavorQuotas struct {
 	// name of this flavor. The name should match the .metadata.name of a
 	// ResourceFlavor. If a matching ResourceFlavor does not exist, the
 	// ClusterQueue will have an Active condition set to False.
-	Name ResourceFlavorReference `json:"name"`
+	// +required
+	Name ResourceFlavorReference `json:"name"` //nolint:kubeapilinter // should not be a pointer
 
 	// resources is the list of quotas for this flavor per resource.
 	// There could be up to 64 resources.
 	// +listType=map
 	// +listMapKey=name
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=64
-	Resources []ResourceQuota `json:"resources"`
+	// +required
+	Resources []ResourceQuota `json:"resources,omitempty"`
```

Comment:
same, why omitempty? And many other places to review like this

Reply:
The linter says to add it or it fails the check.